acceptance/tests/config-entries/config_entries_test.go

```diff
@@ -85,12 +85,12 @@ func TestController(t *testing.T) {
 					// Retry the kubectl apply because we've seen sporadic
 					// "connection refused" errors where the mutating webhook
 					// endpoint fails initially.
-					out, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "apply", "-k", "../fixtures/bases/crds-oss")
+					out, err := k8s.RunKubectlAndGetOutputE(r, ctx.KubectlOptions(r), "apply", "-k", "../fixtures/bases/crds-oss")
 					require.NoError(r, err, out)
-					helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+					helpers.CleanupWithOnFailure(r, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
 						// Ignore errors here because if the test ran as expected
 						// the custom resources will have been deleted.
-						k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "delete", "-k", "../fixtures/bases/crds-oss")
+						k8s.RunKubectlAndGetOutputE(r, ctx.KubectlOptions(r), "delete", "-k", "../fixtures/bases/crds-oss")
```

Comment:
This probably wont execute in the manner you expect. The `CleanupWithOnFailure` method will include the `r` within a closure. As creation succeeded this retry block will not fail its final attempt. When the cleanup method is invoked it will call `r.Failed()` which in this case will always return false and so this will never be run.

I _think_ to correct the behavior we should move the `helpers.CleanupWithOnFailure` call outside of the `retry.Run`